x/evm/keeper/statedb.go

```diff
@@ -86,7 +87,8 @@ func (k *Keeper) SetBalance(ctx sdk.Context, addr common.Address, amount *big.In
 	params := k.GetParams(ctx)
 	coin := k.bankKeeper.GetBalance(ctx, cosmosAddr, params.EvmDenom)
 	balance := coin.Amount.BigInt()
-	delta := new(big.Int).Sub(amount, balance)
+	scaledUpBalance := utils.Convert6To18Decimals(*balance)
+	delta := new(big.Int).Sub(amount, scaledUpBalance)
 	switch delta.Sign() {
```

Comment:
after getting the delta, we'll need to scale it down to 6 decimals for the minting/burning no?
cause otherwise we're minting/burning evmDenom coins with 18 decimals

Reply:
the `amount` already comes in 18 decimals so scaling up the balance and subtracting yields the final correct amounts to be minted burned. You can try it on a local node with the code you game as an example with a direct call send.